homeassistant/components/london_underground/sensor.py

```diff
@@ -96,8 +127,10 @@ def extra_state_attributes(self):
         self.attrs["Description"] = self._description
         return self.attrs
 
-    async def async_update(self):
+    @callback
+    def _handle_coordinator_update(self) -> None:
         """Update the sensor."""
-        await self._data.update()
-        self._state = self._data.data[self.name]["State"]
-        self._description = self._data.data[self.name]["Description"]
+        self._state = self.coordinator.data[self.name]["State"]
+        self._description = self.coordinator.data[self.name]["Description"]
+
+        self.async_write_ha_state()
```

Comment:
Pretty sure you don't need to reimplement this function. You can just use things like `self.coordinator.data[self.name]["State"]` directly in `native_value()` and `extra_state_attributes()`.
Would also let you get rid of `self._state` and `self._description`

Reply:
Not saying you are not correct (to be honest, my current knowledge of this is limited), but I just followed the existing documentation format: https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities

Reply:
I made the changes following your comments, seems to work fine! Thanks!! 🙂